src/main/java/core/basesyntax/dao/FruitDao.java

```diff
@@ -0,0 +1,13 @@
+package core.basesyntax.dao;
+
+import core.basesyntax.model.Fruit;
+import java.util.List;
+import java.util.Optional;
+
+public interface FruitDao {
+    void add(Fruit fruit);
+
+    Optional<Fruit> get(Fruit fruit);
+
+    List<Fruit> getAll();
```

Comment:
```suggestion
    void save(Fruit fruit);

    boolean contains(Fruit fruit);

    List<Fruit> getAll();
```

Reply:
Done

Reply:
Why do we rename get to contains and change return type? In such a case we adjust our program for using Set as our storage. If I remember correctly when using JPA you get the object returned and make manipulations with it and then save it back

Reply:
yes, I suggested this method for you to easier cope with Set as storage

you can leave `Optional<Fruit> get` if you wish